khard/carddav_object.py

```diff
@@ -656,31 +703,57 @@ def _get_private_objects(self):
                     key = self.supported_private_objects[key_index]
                     if key not in private_objects:
                         private_objects[key] = []
-                    private_objects[key].append(child.value)
+                    ablabel = self._get_ablabel(child)
+                    private_objects[key].append(ablabel + (": " if ablabel else "") + child.value)
         # sort private object lists
         for value in private_objects.values():
             value.sort()
         return private_objects
 
     def _add_private_object(self, key, value):
         private_obj = self.vcard.add('X-' + key.upper())
-        private_obj.value = helpers.convert_to_vcard(key, value,
-                                                     ObjectType.string)
+        if isinstance(value, dict):
+            if len(value) > 1:
+                raise ValueError(
+                    "Error: " + key + " must be a string.")
+            label = [i for i in value][0]
+            group_name = self._get_new_group()
+            private_obj.group = group_name
+            private_obj.value = value[label]
+            ablabel_obj = self.vcard.add('X-ABLABEL')
+            ablabel_obj.group = group_name
+            ablabel_obj.value = label
+        else:
+            private_obj.value = helpers.convert_to_vcard(key, value,
+                                                         ObjectType.string)
 
     def _get_webpages(self):
         """
-        :rtype: list(list(str))
+        :rtype: list(str)
         """
         urls = []
         for child in self.vcard.getChildren():
             if child.name == "URL":
-                urls.append(child.value)
+                ablabel = self._get_ablabel(child)
+                urls.append(ablabel + (": " if ablabel else "") + child.value)
         return sorted(urls)
 
     def _add_webpage(self, webpage):
         webpage_obj = self.vcard.add('url')
-        webpage_obj.value = helpers.convert_to_vcard("webpage", webpage,
-                                                     ObjectType.string)
+        if isinstance(webpage, dict):
```

Comment:
This looks like the same code as above. Does it make sense to add a wrapper function, something like `convert_to_vcard_with_ablabel()`?

Reply:
Added as `_add_labelled_object` in #208.